dll/shellext/zipfldr/CZipFolder.hpp

```diff
@@ -0,0 +1,627 @@
+    }
+    STDMETHODIMP EnumSearches(IEnumExtraSearch **ppenum)
+    {
+        DbgPrint("%s\n", __FUNCTION__);
+        return E_NOTIMPL;
+    }
+    STDMETHODIMP GetDefaultColumn(DWORD dwRes, ULONG *pSort, ULONG *pDisplay)
+    {
+        DbgPrint("%s\n", __FUNCTION__);
+        return E_NOTIMPL;
+    }
+    STDMETHODIMP GetDefaultColumnState(UINT iColumn, DWORD *pcsFlags)
+    {
+        if (!pcsFlags || iColumn >= _countof(g_ColumnDefs))
+            return E_INVALIDARG;
+        *pcsFlags = g_ColumnDefs[iColumn].dwDefaultState;
+        return S_OK;
+    }
+    STDMETHODIMP GetDetailsEx(PCUITEMID_CHILD pidl, const SHCOLUMNID *pscid, VARIANT *pv)
+    {
+        DbgPrint("%s\n", __FUNCTION__);
+        return E_NOTIMPL;
+    }
+    // Adapted from CFileDefExt::GetFileTimeString
+    BOOL _GetFileTimeString(LPFILETIME lpFileTime, LPWSTR pwszResult, UINT cchResult)
+    {
+        SYSTEMTIME st;
+
+        if (!FileTimeToSystemTime(lpFileTime, &st))
+            return FALSE;
+
+        size_t cchRemaining = cchResult;
+        LPWSTR pwszEnd = pwszResult;
+        int cchWritten = GetDateFormatW(LOCALE_USER_DEFAULT, DATE_SHORTDATE, &st, NULL, pwszEnd, cchRemaining);
+        if (cchWritten)
+            --cchWritten; // GetDateFormatW returns count with terminating zero
+        else
+            return FALSE;
+        cchRemaining -= cchWritten;
+        pwszEnd += cchWritten;
+
+        StringCchCopyExW(pwszEnd, cchRemaining, L" ", &pwszEnd, &cchRemaining, 0);
+
+        cchWritten = GetTimeFormatW(LOCALE_USER_DEFAULT, 0, &st, NULL, pwszEnd, cchRemaining);
+        if (cchWritten)
+            --cchWritten; // GetTimeFormatW returns count with terminating zero
+        else
+            return FALSE;
+
+        return TRUE;
+    }
+    STDMETHODIMP GetDetailsOf(PCUITEMID_CHILD pidl, UINT iColumn, SHELLDETAILS *psd)
+    {
+        if (iColumn >= _countof(g_ColumnDefs))
+            return E_FAIL;
+
+        psd->cxChar = g_ColumnDefs[iColumn].cxChar;
+        psd->fmt = g_ColumnDefs[iColumn].fmt;
+
+        if (pidl == NULL)
+        {
+            return SHSetStrRet(&psd->str, _AtlBaseModule.GetResourceInstance(), g_ColumnDefs[iColumn].iResource);
+        }
+
+        PCUIDLIST_RELATIVE curpidl = ILGetNext(pidl);
+        if (curpidl->mkid.cb != 0)
+            __debugbreak();
+
+        const ZipPidlEntry* zipEntry = _ZipFromIL(pidl);
+        if (!zipEntry)
+            return E_INVALIDARG;
+
+        WCHAR Buffer[100];
+        bool isDir = zipEntry->ZipType == ZIP_PIDL_DIRECTORY;
+        switch (iColumn)
+        {
+        case 0: /* Name, ReactOS specific? */
+            return GetDisplayNameOf(pidl, 0, &psd->str);
+        case 1: /* Type */
+        {
+            SHFILEINFOA shfi;
+            DWORD dwAttributes = isDir ? FILE_ATTRIBUTE_DIRECTORY : FILE_ATTRIBUTE_NORMAL;
+            ULONG_PTR firet = SHGetFileInfoA(zipEntry->Name, dwAttributes, &shfi, sizeof(shfi), SHGFI_USEFILEATTRIBUTES | SHGFI_TYPENAME);
```

Comment:
"firet"?

Reply:
`FileInfo RETurn` I guess, probably copied that from somwhere.

Reply:
FileInfo-RET